### PR TITLE
Improve error message on schema validation

### PR DESCRIPTION
**Pull request by Czaki**
When renaming the package I use `_` instead of `-` in the new name and error message does not allow me to understand where is the source of the problem. 

In this PR I improve the error message to have information where is a problem.

### CODE REVIEW

**Review by tlambert03 — APPROVED** [review]
looks good thanks.  while we're here, I actually think the phrase "Manifest name" that we previously used is a little ambiguous: it could be interpreted as the name of the manifest file, yeah?  What do you think of 'manifest.name' must match package name or "the name field in the manifest must match the name of the package"?

**Comment by Czaki** [comment]
> the name field in the manifest must match the name of the package

I like this, but with provided name or information that `_` are replaced with `-`.

**Comment by tlambert03** [comment]
> > the name field in the manifest must match the name of the package
> 
> I like this, but with provided name or information that `_` are replaced with `-`.

yeah, leave in the bit you added

File: npe2/manifest/schema.py [hunk 1]
```diff
@@ -321,7 +321,9 @@ def _from_entrypoint(
                     meta = PackageMetadata.from_dist_metadata(distribution.metadata)
                     mf.package_metadata = meta
 
-                    assert mf.name == meta.name, "Manifest name must match package name"
+                    assert (
+                        mf.name == meta.name
+                    ), f"Manifest name ({mf.name}) must match package name ({meta.name})"
```
Comment (tlambert03) [inline npe2/manifest/schema.py:326]:
```suggestion
                    assert (
                        mf.name == meta.name
                    ), f"The name field in the manifest ({mf.name!r}) must match package name ({meta.name!r})"
```

**Pull request merged**